src/client/classroom/session/chatbox/MessageCompose.jsx

```diff
@@ -1,34 +1,109 @@
-import React from 'react';
-import { InputGroup, FormControl, Button } from 'react-bootstrap';
+import React, { useEffect } from 'react';
+import {
+  InputGroup, FormControl, Button, Form
+} from 'react-bootstrap';
 import { useStates, bindState } from '../../../hooks/useStates.js';
+import { useRealtime } from '../../../contexts/RealtimeProvider.jsx';
+import { useDataStore } from '../../../contexts/DataStoreProvider.jsx';
 
 // TODO: reply to question
 
 export default function MessageCompose({ onCreateQuiz }) {
-  const data = useStates({
-    message: ''
+  const { sendMessage } = useRealtime();
+  const { data } = useDataStore();
+
+  const messageData = useStates({
+    message: '',
+    information: null
   });
 
-  const onSend = () => {
-    console.log(data.message);
+  const onSend = async () => {
+    if (!messageData.message) return;
+    console.log('Message object: ', { message: messageData.message, information: messageData.information });
+    try {
+      await sendMessage(messageData.message, messageData.information);
+    } catch (ex) {
+      console.log('send message error: ', ex);
+    }
```

Comment:
Since `sendMessage` is written in the `return new Promise` style, you should await the function call and use `try catch` to get the error message in case the server returns an error